### DISCUSSION TITLE
How to make a field disabled only on edit form, but only of a certian condition is true

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#14388] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I know I can do this
~~~
->disabled($condition)
~~~
And also this
~~~
->disabledOn('edit')
~~~

But I want to disable the field only on edit and only if the records book is not current...
How can I achieve the disabled only on edit but together with a condition

**Comment by leandrocfe** [comment]
Inject `$operation`

https://filamentphp.com/docs/3.x/forms/advanced/#injecting-the-current-form-operation

```php
->disabled(function(string $operation) :bool {
    if($operation === 'edit') {
        //...
    }
})
```

**Reply by ariesdg** [reply]
Hi @leandrocfe 

Is it safe to use this disabled if you want conditionally prohibit user from editing certain field?

**Reply by leandrocfe** [reply]
You can also add [dehydrated](https://filamentphp.com/docs/3.x/forms/advanced/#preventing-a-field-from-being-dehydrated)
